src/mumble/API_v_1_x_x.cpp

```diff
@@ -1093,6 +1172,80 @@ void MumbleAPI::requestSetLocalUserComment_v_1_0_x(mumble_plugin_id_t callerID,
 	EXIT_WITH(MUMBLE_STATUS_OK);
 }
 
+void MumbleAPI::requestLinkChannels_v_1_3_x(mumble_plugin_id_t callerID, mumble_channelid_t *channelList,
+											std::size_t channelCount, std::shared_ptr< api_promise_t > promise) {
+	if (QThread::currentThread() != thread()) {
+		// Invoke in main thread
+		QMetaObject::invokeMethod(this, "requestLinkChannels_v_1_3_x", Qt::QueuedConnection,
+								  Q_ARG(mumble_plugin_id_t, callerID), Q_ARG(mumble_channelid_t *, channelList),
+								  Q_ARG(std::size_t, channelCount), Q_ARG(std::shared_ptr< api_promise_t >, promise));
+
+		return;
+	}
+	api_promise_t::lock_guard_t guard = promise->lock();
+	if (promise->isCancelled()) {
+		return;
+	}
+
+	VERIFY_PLUGIN_ID(callerID);
+
+	if (channelCount < 2) {
+		EXIT_WITH(MUMBLE_STATUS_OK);
+	}
```

Comment:
Hm. I'm unsure about whether or not this should emit an error in this case. My first instinct would be to error if the set of provided channel IDs has less than two channels in it. On the other hand the task of this function becomes a no-op in this case and so the function can therefore conclude that it did its job as described :thinking: 

Still, if this occurs I would bet that in 99.9% of cases users would appreciate this being reported as an error.
Additionally, what if only a single channel ID is provided but that ID is invalid. Normally, any invalid channel ID would result in an error, whereas here the function would return and claim everything is okay...

@davidebeatrici @Hartmnt what are your opinions on this matter?